src/player.cpp

```diff
@@ -322,10 +316,31 @@ void Player::Update(bool update_scene) {
 		}
 	}
 
+#ifdef EMSCRIPTEN
+	Graphics::Draw();
+#else
+	cur_time = (double)DisplayUi->GetTicks();
+	if (cur_time < next_frame) {
+		Graphics::Draw();
+		cur_time = (double)DisplayUi->GetTicks();
+		// Don't use sleep when the port uses an external timing source
```

Comment:
Would the emscripten way not suffice for `libretro` too?

Reply:
no, because this way you get frameskip. The emscripten way will slow down the engine when you are < 60 FPS.
I wanted to use the same technique for emscripten but GetTicks is a bit broken there and doesn't work correctly.